settings: Fixed spellchecker language selection

cyrilchukwuebuka opened the pull request:
---

<!--
Remove the fields that are not appropriate
Please include:
-->

**What's this PR do?**
Fixed issue #1286 

**Any background context you want to provide?**
I noticed using Tagify on the input fields makes the section behave in a weird manner.

So in [f2972c0](https://github.com/zulip/zulip-desktop/pull/1290/commits/f2972c0800b1b5c35361eb4a3da7491ee92f79f0) I replaced it with a drop-down list of system-available language spellcheckers for users to select from without having to guess or input the wrong spellcheck language hereby making it easier for users to obviously know the list of available/supported languages to pick from thereby improving the user experience on the platform.

**Screenshots?**
![spellchecker_image](https://user-images.githubusercontent.com/41521163/228079402-2cbd3067-8cf3-40c7-8d77-992e9a542a6d.PNG)

**You have tested this PR on:**

- [x] Windows
- [ ] Linux/Ubuntu
- [ ] macOS

timabbott commented:
@cyrilchukwuebuka please see https://zulip.readthedocs.io/en/latest/contributing/reviewable-prs.html and clean up the commit history; this PR is not ready for review as written.

I notice one immediate visual issue: the dropdown's text look way too close to the left edge. Please spend some time polishing this before requesting a next review.

cyrilchukwuebuka commented:
Okay, I will do just that
Thanks for the review @timabbott

cyrilchukwuebuka committed:
settings: Fixed spellchecker language selection

cyrilchukwuebuka commented:
@timabbott, I have cleaned up the commit history and also added padding to the left edge of the dropdown text
I would appreciate it if you could review these changes once again

andersk commented:
Thanks for the report @cyrilchukwuebuka! I’ve opted to fix the old UI in 0cb729701735711802b459bcdb63c2ea7ea83308 rather than switching to a different UI.
